source/Mlos.Model.Services/Spaces/Hypergrids.cs

```diff
@@ -44,23 +53,14 @@ public class Hypergrid : IHypergrid
         /// </summary>
         /// <param name="jsonString"></param>
         /// <returns></returns>
+        [System.Obsolete("JSON serialization is deprecated. Please use OptimizerServiceEncoderDecoder instead.", true)]
         public static Hypergrid FromJson(string jsonString)
         {
             Hypergrid hypergrid = (jsonString != null) ? JsonSerializer.Deserialize<Hypergrid>(jsonString, JsonSerializerOptions) : null;
 
             return hypergrid;
         }
 
-        /// <summary>
-        /// Internal structure describing subgrid joins.
-        /// </summary>
-        internal struct SubgridJoin
-        {
-            internal IDimension OnExternalJoin { get; set; }
-
-            internal Hypergrid Subgrid;
-        }
-
         // This is for the benefit of the JsonSerializer.

```

Comment:
outdated comment

Reply:
[zaporter/grpc 630599a6] Improved readability of if (var is Type varname) and removed old comment